src/lib/djerba/plugins/fusion/plugin.py

```diff
@@ -67,22 +80,155 @@ def sort_by_actionable_level(row):
+        output_tsv_path = os.path.join(output_dir, 'fusion_blurb_urls.tsv')
+        with open(output_tsv_path, 'w', newline='') as tsvfile:
+            writer = csv.writer(tsvfile, delimiter='\t')
+            writer.writerow(['Fusion', 'Whizbam URL'])
+            writer.writerows(fusion_url_pairs)
+
         data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)
         data[core_constants.RESULTS] = results
         data[core_constants.MERGE_INPUTS]['gene_information_merger'] = gene_info
         data[core_constants.MERGE_INPUTS]['treatment_options_merger'] = treatment_opts
         return data
 
+    def process_fusion(self, config, fusion, tsv_file_path, json_template_path, output_dir):
+        wrapper = self.get_config_wrapper(config)
+
+        # Validate and parse the fusion format
+        match = re.match(r"(.+)::(.+)", fusion)
+        if not match:
+            raise FusionProcessingError(f"No valid fusion found for {fusion}. Ensure the format is gene1::gene2.")
+        gene1, gene2 = match.groups()
+
+        # Find breakpoints in the ARRIBA TSV file
+        breakpoint1, breakpoint2 = self.find_breakpoints(tsv_file_path, gene1, gene2)
+        if not (breakpoint1 and breakpoint2):
+            raise FusionProcessingError(f"No matching fusion found in the TSV file ({tsv_file_path}) for {fusion}.")
+
+        # Format breakpoints
+        formatted_breakpoint1 = self.format_breakpoint(breakpoint1)
+        formatted_breakpoint2 = self.format_breakpoint(breakpoint2)
+
+        # Load the JSON template
+        with open(json_template_path, 'r') as json_file:
+            data = json.load(json_file)
+
+        # Update the JSON with the formatted breakpoints
+        data['locus'] = [formatted_breakpoint1, formatted_breakpoint2]
+
+        project_id = wrapper.get_my_string(core_constants.PROJECT)
+        tumour_id = wrapper.get_my_string(core_constants.TUMOUR_ID)
+        whizbam_project_id = wrapper.get_my_string(fc.WHIZBAM_PROJECT)
+        data['tracks'][1]['name'] = tumour_id
+
+        # Define file patterns
+        bam_project_path = f"{core_constants.WHIZBAM_PATTERN_ROOT}/{project_id}/RNASEQ/{tumour_id}.bam"
+        bai_project_path = f"{core_constants.WHIZBAM_PATTERN_ROOT}/{project_id}/RNASEQ/{tumour_id}.bai"
+        bam_whizbam_path = f"{core_constants.WHIZBAM_PATTERN_ROOT}/{whizbam_project_id}/RNASEQ/{tumour_id}.bam"
+        bai_whizbam_path = f"{core_constants.WHIZBAM_PATTERN_ROOT}/{whizbam_project_id}/RNASEQ/{tumour_id}.bai"
+
+        # Resolve BAM file
+        bam_file, bam_project = None, None
+        if os.path.isfile(bam_project_path):
+            bam_file, bam_project = bam_project_path, project_id
+        elif os.path.isfile(bam_whizbam_path):
+            bam_file, bam_project = bam_whizbam_path, whizbam_project_id
+        else:
+            warnings.warn(f"BAM file not found for {project_id}. Try adjusting whizbam_project_id in config file")
```

Comment:
Use `self.logger.warning` unless there is a good reason to do otherwise -- not sure what benefit there is to using `warnings.warn` instead?